src/snowflake/connector/result_set.py

```diff
@@ -63,41 +63,27 @@ def result_set_iterator(
     """
 
     with ThreadPoolExecutor(prefetch_thread_num) as pool:
-        # Fill up window
-
         logger.debug("beginning to schedule result batch downloads")
-
         for _ in range(min(prefetch_thread_num, len(unfetched_batches))):
             logger.debug(
                 f"queuing download of result batch id: {unfetched_batches[0].id}"
             )
             unconsumed_batches.append(
                 pool.submit(unfetched_batches.popleft().create_iter, **kw)
             )
-
         yield from first_batch_iter
-
+        while unfetched_batches:
+            logger.debug(
+                f"queuing download of result batch id: {unfetched_batches[0].id}"
+            )
+            future = pool.submit(unfetched_batches.popleft().create_iter, **kw)
+            unconsumed_batches.append(future)
+        _, _ = wait(unconsumed_batches, return_when=ALL_COMPLETED)
```

Comment:
does this await all the chunks to complete? let's assume there are 100 chunks in total, which means the program will hang here until all chunks are downloaded?

Reply:
Yes, but it is the same for original code, the original code also wait for all chunks to download

Reply:
I think in the previous logic is downloading chunk 1 -> yield the result -> downloading chunk 2 -> ...
while the current logic is downloading chunk 1,2 ... n, -> yield the result one by one.

so technically the code change would impact the timing when users get the first chunk of data?

Reply:
all the usage of result_set_iterator is by calling list(...), not impacting perf